controllers/construct/build_statefulset_test.go

```diff
@@ -87,6 +88,7 @@ func assertStatefulSetIsBuiltCorrectly(t *testing.T, mdb mdbv1.MongoDBCommunity,
 	assert.Equal(t, mongodbDatabaseServiceAccountName, sts.Spec.Template.Spec.ServiceAccountName)
 	assert.Len(t, sts.Spec.Template.Spec.Containers[0].Env, 4)
 	assert.Len(t, sts.Spec.Template.Spec.Containers[1].Env, 1)
+	assert.Equal(t, podtemplatespec.DefaultPodSecurityContext(), *sts.Spec.Template.Spec.SecurityContext)
```

Comment:
Is it possible that `sts.Spec.Template.Spec.SecurityContext` is nil here? It's good to have require.NotNil before dereferencing if it's a possibility. If it's nil the panic is polluting the logs.

Reply:
In this  unit test, it's not. However, you have a very good point, I should add a test testing a managed security (so that it will be nil). Good spot!

Reply:
Added more tests in this area (and a requested nil/non-nil check).